lib/ClangImporter/ImportDecl.cpp

```diff
@@ -345,6 +345,23 @@ static bool isNSDictionaryMethod(const clang::ObjCMethodDecl *MD,
   return true;
 }
 
+static void forEachDistinctName(
+    ClangImporter::Implementation &impl, const clang::NamedDecl *decl,
+    llvm::function_ref<void(ImportedName, ImportNameVersion)> action) {
+  using ImportNameKey = std::pair<DeclName, EffectiveClangContext>;
+  SmallVector<ImportNameKey, 8> seenNames;
+  forEachImportNameVersionFromCurrent(impl.CurrentVersion,
+                                      [&](ImportNameVersion nameVersion) {
+    // Check to see if the name is different.
+    ImportedName newName = impl.importFullName(decl, nameVersion);
+    ImportNameKey key(newName, newName.getEffectiveContext());
```

Comment:
Yes, Swift 2 and Swift 3 might have the same imported name, in which case we don't actually want to import the declaration twice.